vyper/ast/folding.py

```diff
@@ -176,16 +176,36 @@ def replace_user_defined_constants(vyper_module: vy_ast.Module) -> int:
 
         # Extract type definition from propagated annotation
         constant_annotation = node.get("annotation.args")[0]
-        type_ = (
-            get_type_from_annotation(constant_annotation, DataLocation.UNSET)
-            if constant_annotation
-            else None
-        )
+        try:
```

Comment:
Added a try/except to handle struct constants, which are user-defined and throws a `UnknownType` exception when executing `get_type_from_annotation`.

Reply:
is the issue that struct resolution happens before constant folding?

Reply:
oh, i see why this works -- struct literals have unambiguous type

Reply:
yea, the type (for constant propagation) cannot be derived for struct literals using `get_type_from_annotation` (and is also unnecessary), so we catch the exception and set it to `None` instead.